docs/source/newsfragments/2720.change.rst

```diff
@@ -0,0 +1 @@
+``cocotb.handle.ConstantObject`` was removed. Constant value objects are now returned as their corresponding :class:`simulator value object <cocotb.handle.NonHierarchyObject>` subtype (e.g. integers in VHDL and Verilog are now :class:`~cocotb.handle.IntegerObject`\ s). You can check if the value object is constant using the :meth:`~cocotb.handle.NonHierarchyObject.is_const` method.
```

Comment:
`NonHierarchyObject.is_const()` doesn't show up in the docs and this link doesn't work.

Reply:
`WARNING: Failed to get a method signature for cocotb.handle.NonHierarchyObject.is_const: <cocotb.utils.lazy_property object at 0x7f413f993880> is not a callable object`